benchmarks/set_http_meta/scenario.py

```diff
@@ -0,0 +1,132 @@
+    "image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
+    "HTTP_SEC_FETCH_SITE": "none",
+    "HTTP_SEC_FETCH_MODE": "navigate",
+    "HTTP_SEC_FETCH_USER": "?1",
+    "HTTP_SEC_FETCH_DEST": "document",
+    "HTTP_ACCEPT_ENCODING": "gzip, deflate, br",
+    "HTTP_ACCEPT_LANGUAGE": "en-US,en;q=0.9",
+    "HTTP_COOKIE": "Pycharm-45729245=449f1b16-fe0a-4623-92bc-418ec418ed4b; Idea-9fdb9ed8="
+    "448d4c93-863c-4e9b-a8e7-bbfbacd073d2; csrftoken=cR8TVoVebF2afssCR16pQeqHcxA"
+    "lA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL; _xsrf=2|d4b85683|7e2604058ea673d12dc6604f"
+    '96e6e06d|1635869800; username-localhost-8888="2|1:0|10:1637328584|23:username-loca'
+    "lhost-8888|44:OWNiOTFhMjg1NDllNDQxY2I2Y2M2ODViMzRjMTg3NGU=|3bc68f938dcc081a9a02e51660"
+    '0c0d38b14a3032053a7e16b180839298e25b42"',
+    "wsgi.input": BytesIO(),
+    "wsgi.url_scheme": "http",
+}
+
+COOKIES = {"csrftoken": "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL"}
+
+DATA_GET = dict(
+    method="GET",
+    url="http://localhost:8888{}".format(PATH),
+    status_code=200,
+    status_msg="OK",
+    query=b"key1=value1&key2=value2&token=cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL",
+    parsed_query={
+        "key1": "value1",
+        "key2": "value2",
+        "token": "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL",
+    },
+    request_headers=COMMON_DJANGO_META,
+    response_headers=COMMON_DJANGO_META,
+    retries_remain=0,
+    raw_uri="http://localhost:8888{}?key1=value1&key2=value2&token="
+    "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL".format(PATH),
+    request_cookies=COOKIES,
+    request_path_params={"id": 1},
+)
+
+DATA_POST = dict(
+    method="GET",
+    url="http://localhost:8888/test-benchmark",
+    status_code=200,
+    status_msg="OK",
+    query=b"key1=value1&key2=value2",
+    parsed_query={"key1": "value1", "key2": "value2"},
+    request_headers=COMMON_DJANGO_META,
+    response_headers=COMMON_DJANGO_META,
+    retries_remain=0,
+    raw_uri="http://127.0.0.1:8081/test/1/?key1=value1&key2=value2",
+    request_cookies=COOKIES,
+    request_path_params={"id": 1},
+    request_body=None,
+)
+
+
+class SetHttpMeta(bm.Scenario):
+    allenabled = bm.var_bool()
+    useragentkey = bm.var_bool()
+    useragentvariants = bm.var_bool()
+
+    def run(self):
+        # run scenario to also set tags on spans
+        if self.allenabled:
+            config = Config(lambda: True)
+        else:
+            config = Config(lambda: False)
+
+        data = copy.deepcopy(DATA_GET)
+
+        user_agent_keys = []
+        if self.useragentkey:
+            user_agent_keys = ["HTTP_USER_AGENT"]
+            if self.useragentvariants:
+                user_agent_keys = ["HTTP_USER_AGENT", "USER_AGENT", "UserAgent", "User-Agent", "user-agent"]
+
+        def bm(loops):
+            for _ in range(loops):
+                for user_agent_key in user_agent_keys:
```

Comment:
I wonder if these should be different variants?

If we run them all in one benchmark will it just show case the average case instead of highlighting the best v worse case scenarios?

e.g. if `HTTP_USER_AGENT` is always the fastest to lookup because we check it first, vs `user-agent` being the slowest because it is the last... for Django META/WSGI environ it'll always be fast, but for a framework only provides `user-agent` it'll always be slow.

Reply:
In general, I agree with @brettlangdon though would avoid adding more variants than we expect to yield meaningful differences.